src/common/os/win32/os_utils.cpp

```diff
@@ -560,6 +561,65 @@ void getUniqueFileId(HANDLE fd, UCharBuffer& id)
 		   sizeof(file_info.nFileIndexLow));
 }
 
+void setDefaultAffinity()
+{
+	HANDLE hCurrProc = GetCurrentProcess();
+
+	DWORD_PTR procMask, sysMask, newMask;
+	GetProcessAffinityMask(hCurrProc, &procMask, &sysMask);
+
+	// Don't change affinity if it was set by caller process
+	if (procMask != sysMask)
+		return;
+
+	newMask = sysMask;
+
+	DWORD len = 0;
+
+	if (!GetLogicalProcessorInformationEx(RelationProcessorCore, nullptr, &len))
+	{
+		if (GetLastError() != ERROR_INSUFFICIENT_BUFFER)
+			return;
+	}
+
+ 	HalfStaticArray<UCHAR, 1024> buff;
```

Comment:
Don't you need this variable aligned as `PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX`? (using third parameter of `HalfStaticArray`)?

Reply:
It is an usual requirement for WinAPI and providing misaligned pointer can end up in ERROR_NOACCESS returned. As an example GetTokenInformation() does that though it is not documented anywhere and for ReadDirectoryChangesEx() it is written explicitly.

Reply:
Not documented == not usual. 
Anyway, I check with 1, 2, 4 and 8 bytes aligned pointers and found that 1 and 2 indeed returns `ERROR_NOACCESS`. 
Note, `alignof(SYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX) == 8`.
Also, `InlineStorage<>` by default returns 8-byte aligned pointer.

Reply:
Fixed.

Reply:
In this case function's parameter is declared as a pointer to struct so "not documented" == "compiler default ABI". If the parameter had been declared as PVOID that would indeed might vary.

Reply:
> Also, `InlineStorage<>` by default returns 8-byte aligned pointer.

It does not.